ambry-router/src/main/java/com/github/ambry/router/TtlUpdateManager.java

```diff
@@ -160,8 +160,13 @@ void handleResponse(ResponseInfo responseInfo) {
         RouterUtils.extractResponseAndNotifyResponseHandler(responseHandler, routerMetrics, responseInfo,
             TtlUpdateResponse::readFrom, TtlUpdateResponse::getError);
     RequestInfo routerRequestInfo = responseInfo.getRequestInfo();
-    int correlationId = ((TtlUpdateRequest) routerRequestInfo.getRequest()).getCorrelationId();
+    int correlationId = routerRequestInfo.getRequest().getCorrelationId();
     TtlUpdateOperation ttlUpdateOperation = correlationIdToTtlUpdateOperation.remove(correlationId);
+    if (ttlUpdateOperation == null) {
+      LOGGER.warn("No TtlUpdateOperation found for correlation id: {}", correlationId);
```

Comment:
Can we do this same logging in the else of line 187 below?

Reply:
updated.